File: insertion sort/insertion_sort.py
def insertion_sort(array) :
    for x in range(1,len(array)) :
        y = x
        temp = array[x]
        while y > 0 and temp < array[y-1] :
            array[y] = array[y-1]
            y -= 1
        array[y] = temp
    return array

array = [1,-1,9,0,-3,6,10,20,67,-9,8,8]
sorted_array = insertion_sort(array)
print("Array after sorting with insertion sort algorithm :- ",sorted_array)

File: insertion sort/insertion_sort.cpp
#include <iostream>

using namespace std;

template <class T>

void Print(T& vec, int n, string s){

    cout << s << ": [" << flush;

    for (int i=0; i<n; i++){
        cout << vec[i] << flush;
        if (i < n-1){
            cout << ", " << flush;
        }
    }

    cout << "]" << endl;
}

void InsertionSort(int A[], int n){

    for (int i=1; i<n; i++){
        int j = i-1;
        int x = A[i];

        while (j>-1 && A[j] > x){
            A[j+1] = A[j];
            j--;
        }
        
        A[j+1] = x;
    }
}

int main() {

    int A[] = {1,-1,9,0,-3,6,10,20,67,-9,8,8};

    InsertionSort(A, sizeof(A)/sizeof(A[0]));
    Print(A, sizeof(A)/sizeof(A[0]), "Array after sorting with insertion sort algorithm :- ");

    return 0;
}
